cpp/include/cudf/table/row_operators.cuh

```diff
@@ -345,6 +345,41 @@ class row_lexicographic_comparator {
     CUDF_EXPECTS(_lhs.num_columns() == _rhs.num_columns(), "Mismatched number of columns.");
   }
 
+  inline __device__ weak_ordering struct_compare(column_device_view const& lhs,
+                                                 column_device_view const& rhs,
+                                                 null_order null_precedence,
+                                                 size_type lhs_index,
+                                                 size_type rhs_index) const noexcept
+  {
+    if (has_nulls) {
+      bool const lhs_is_null{lhs.nullable() and lhs.is_null(lhs_index)};
+      bool const rhs_is_null{rhs.nullable() and rhs.is_null(rhs_index)};
```

Comment:
I know this was copied from elsewhere, but this is actually redundant. `is_null` already checks for `nullable()`.  When I originally wrote `is_null` it didn't check for `nullable()`. That's what `is_null_nocheck` is for. 
```suggestion
      bool const lhs_is_null{lhs.is_null(lhs_index)};
      bool const rhs_is_null{rhs.is_null(rhs_index)};
```